backend/src/services/tenant/tenant.entity.ts

```diff
@@ -0,0 +1,47 @@
+import { getRandomId } from '~/helpers/helpers';
+
+class Tenant {
+  public id: string;
+  public name: string;
+  public createdAt: Date;
+
+  private constructor({
+    id,
+    name,
+    createdAt,
+  }: {
+    id: string;
+    name: string;
+    createdAt: Date;
+  }) {
+    this.id = id;
+    this.name = name;
+    this.createdAt = createdAt;
+  }
+
+  public static initialize({
+    id,
+    name,
+    createdAt,
+  }: {
+    id: string;
+    name: string;
+    createdAt: Date;
+  }): Tenant {
+    return new Tenant({
+      id,
+      name,
+      createdAt,
+    });
+  }
+
+  public static createNew({ name }: { name: string }): Tenant {
```

Comment:
```suggestion
  public static createNew({ name }: { name: string }): Tenant {
```
create a type for it. don't use object literal as a type